src/components/MovieList/MovieList.jsx

```diff
@@ -15,12 +23,25 @@ class MovieList extends Component {
     };
   }
 
-  componentDidMount() {
-    API.getMovies().then((movies) => {
-      this.setState({
-        movies: [...movies.data],
+  componentWillReceiveProps({ location }) {
+    if (/search/.test(location.pathname)) {
+      const nextSearchParams = new URLSearchParams(location.search);
+      const prevSearchParams = new URLSearchParams(this.props.location.search);
+      if (location.pathname === this.props.location.pathname &&
+        nextSearchParams.get('searchBy') === prevSearchParams.get('searchBy') &&
+        nextSearchParams.get('searchFor') === prevSearchParams.get('searchFor')) {
+        console.log('no update');
```

Comment:
Remove console.log. And I would prefer to compare search params as strings and them convert them to object

Reply:
need some clarifications